p2p/src/simulated/ingress.rs

```diff
@@ -205,6 +225,25 @@ pub struct Control<P: PublicKey> {
     sender: mpsc::UnboundedSender<Message<P>>,
 }
 
+impl<P: PublicKey> Control<P> {
+    /// Register the communication interfaces for the peer over a given [Channel].
+    pub async fn register_comms(
```

Comment:
See comment about keeping this `register` and changing peer set changes to `update`